pkg/domain/transformer.go

```diff
@@ -0,0 +1,46 @@
+package domain
+
+import (
+	logger "github.com/sirupsen/logrus"
+	"strconv"
+	"strings"
+)
+
+func parseTransMap(s string) (map[float64]float64, error) {
+	m := make(map[float64]float64)
+	for _, kv := range strings.Split(s, ",") {
+		kvSplit := strings.Split(kv, ":")
+		if len(kvSplit) == 2 {
+			key, errKey := strconv.ParseFloat(kvSplit[0], 64)
+			value, errVal := strconv.ParseFloat(kvSplit[1], 64)
+
+			if errKey != nil {
+				logger.Errorf("Error converting key to float: %s", errKey)
+				return nil, errKey
+			}
+			if errVal != nil {
+				logger.Errorf("Error converting value to float: %s", errVal)
+				return nil, errVal
+			}
+
+			m[key] = value
+		}
+	}
+	return m, nil
+}
+
+func Transform(telem *TelemetryMessage, route *TransformerValue) (*TelemetryMessage, error) {
```

Comment:
this takes in a telem pointer but also returns a telem pointer. it should either
1. take in a point and only return error( alter inline)  
2. take in a non-pointer and return a non-pointer(alter and return)

rename route param. 
Does the router param need to be a pointer?